File: C/output/LISTA 4/ativ 01.c
#include <stdio.h>
#include <stdlib.h>
#include <windows.h> // acentuação
#include <math.h>

//------------------VALIDAÇÃO DA NOTA------------------//

float lerNota( int cont){

    float nt; 

    do{
        printf("Digite sua %d° nota: ", cont + 1);
        scanf("%f", &nt);

    }while (nt <= 0 || nt > 10);

    return nt; 
}

//------------------LER AS TRÊS NOTAS------------------//

float ler3notas (int cont){

    float somaNotas, nt;
    somaNotas = 0;  

    while (cont < 3)
    {
        nt = lerNota(cont); 
        cont = cont + 1; 
        somaNotas = somaNotas + nt; 
    }
    
    return somaNotas; 

}

//------------------CALCULAR A MÉDIA E IMPRIMIR A MENSAGEM------------------//

float calcMedia (float somarNotas){

    float media; 
    media = somarNotas / 3; 

    if (media >= 7){

        printf("Média final: %.2f\n", media); 
        printf("Aprovado");

    }else if (media < 6){   

        printf("Média final: %.2f\n", media);  
        printf("Reprovado");

    }else if(media >= 6 && media < 7){

        printf("Média final: %.2f\n", media);  
        printf("Prova Final"); 
    }
    
    return media; 
}

//------------------LER E VÁLIDAR A MATRÍCULA------------------// 

float lerMat(){

    int matricula; 

    do{
        printf("Digite sua matrícula: ");
        scanf("%d", &matricula);

    }while (matricula <= 0);

    return matricula; 
}

//------------------PROGRAMA PRINCIPAL------------------//

int main()
{ 
    SetConsoleOutputCP(65001); 


    float somarNotas;
    int matricula, cont; 
    cont = 0; 

    printf("Olá, seja bem vindo ao sistema de cálculo de média de alunos\n"); 

    matricula = lerMat();
    printf("Aluno %d\n", matricula);

    somarNotas = ler3notas(cont); 
    calcMedia(somarNotas);

    return 0; 
}

File: PYTHON/LISTA 6/ativ01.py
# Atividade 1 - Lista 6

def lerNota():

    nota = float(input("Digite a nota do aluno: "))
    
    while nota < 0 or nota >10:
        print("Nota inválida!")
        nota = float(input("Digite a nota do aluno: "))

    return nota

#--------------------------------------------------------#

def lerNome():

    nome = input("Digite o nome do aluno: ")

    return nome

#--------------------------------------------------------#

def Calmedia(n1, n2, n3):

    total = (n1 + n2 + n3)
    media = total/3

    return media, total

#--------------------------------------------------------#

def decrescente(n1,n2,n3, total):

#maior nota
  
    if n1>=n2 and n1>=n3: 
        maiorNota = n1

    elif n2>=n1 and n2>=n3:
        maiorNota = n2

    else:
        maiorNota = n3

#menor nota

    if n1<=n2 and n1<=n3:
        menorNota = n1

    elif n2<=n1 and n2<=n3: 
        menorNota = n2

    else:
        menorNota = n3
    
    meio = total - menorNota - maiorNota

    return maiorNota, menorNota, meio

#--------programa principal----------#

nome = lerNome()

n1 = lerNota()
n2 = lerNota()
n3 = lerNota()

media, total = Calmedia(n1, n2, n3)

maiorNota, menorNota, meio = decrescente(n1, n2, n3, total)

print("A média do aluno(a) %s foi %.2f" %(nome, media))

if media >= 7: 
    print("Aluno aprovado!")

elif media < 6:
    print("Aluno reprovado!")

else:
    print("Prova final!")

print("Suas notas foram: %.2f, %.2f,%.2f" %(maiorNota, meio, menorNota))

